src/srr-mapping.ts

```diff
@@ -70,6 +71,40 @@ export function handleTransfer(event: TransferEvent): void {
   }
 }
 
+export function handleApproval(event: ApprovalEvent): void {
+  let timestampMillis = eventUTCMillis(event)
+  let srrId = event.params.tokenId.toString()
+  let owner = event.params.owner.toHexString()
+  let approved = event.params.approved.toHexString()
+  
+  log.info('Approval for {}', [srrId])
+  log.info('owner: {}', [owner])
+  log.info('approved: {}', [approved])
+  
+
+  let srr = SRR.load(srrId)
+  if (srr == null) {
+    log.error('received event for unknown SRR: {}', [srrId])
+    return
+  }
+
+  // Create new approval
+  let approvalId = crypto.keccak256(
+    ByteArray.fromUTF8(
+      srrId.toString()
```

Comment:
@noyonthe1 

you will get a duplicate exception here for a 2nd approval. 

actually in the schema you have a comment on ID `  id: ID!                      # keccak256(tokenId,timestamp)` and that wlil fix it actually. add the timestamp into the hash.

Reply:
oh ya, right, it's my mistake, I was about to add a timestamp then forgot after getting the error into the event. thanks for mentioning it. @chatch

Reply:
timestamp added into this commit https://github.com/startbahn/startrail-subgraph/pull/48/commits/333f2c6806a94fcf40a7ca316aded262aff9eaa7